app/src/main/java/com/omkarmoghe/pokemap/controllers/map/LocationManager.java

```diff
@@ -78,12 +79,14 @@ public void onConnected(@Nullable Bundle bundle) {
                         @Override
                         public void onConnectionSuspended(int i) {
 
+                            notifyLocalizationFailed();
                         }
                     })
                     .addOnConnectionFailedListener(new GoogleApiClient.OnConnectionFailedListener() {
                         @Override
                         public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
 
+                            notifyLocalizationFailed();
```

Comment:
Recommend sending a parameter of the error object back up the chain, so the receiver knows the issue. Would need to build up on the handling of said issues as we move forwards.